fdbserver/workloads/AuthzSecurity.actor.cpp

```diff
@@ -0,0 +1,409 @@
+		state Transaction tr(cx, self->tenant);
+		state Key key = self->randomString();
+		state Value value = self->randomString();
+		state Value newValue = self->randomString();
+		self->setAuthToken(tr, self->signedToken);
+		wait(self->setKeyValue(&tr, key, value));
+		// refresh key location cache to extract tenant prefix
+		KeyRangeLocationInfo loc = wait(self->extractCachedLocation(&tr, cx, self->tenant, key));
+		if (positive) {
+			// Expected to succeed, may occasionally fail
+			Optional<Error> outcome = wait(self->tryCommit(
+			    self->tenant, self->signedToken, key, newValue, tr.getCommittedVersion(), cx, loc, self));
+			if (!outcome.present()) {
+				++self->crossTenantCommitPositive;
+			} else if (outcome.get().code() == error_code_permission_denied) {
+				TraceEvent(SevError, "AuthzSecurityError")
+				    .detail("Case", "CrossTenantGetDisallowed")
+				    .detail("Subcase", "Positive")
+				    .log();
+			}
+		} else {
+			Optional<Error> outcome = wait(self->tryCommit(
+			    self->tenant, self->signedTokenAnotherTenant, key, newValue, tr.getCommittedVersion(), cx, loc, self));
+			if (!outcome.present()) {
+				TraceEvent(SevError, "AuthzSecurityError")
+				    .detail("Case", "CrossTenantGetDisallowed")
+				    .detail("Subcase", "Negative")
+				    .log();
+			} else if (outcome.get().code() == error_code_permission_denied) {
+				++self->crossTenantCommitNegative;
+			}
+		}
+		return Void();
+	}
+
+	ACTOR Future<Void> testPublicNonTenantRequestsAllowedWithoutTokens(Database cx, AuthzSecurityWorkload* self) {
+		state Transaction tr(cx, self->tenant);
+		loop {
+			try {
+				Version version = wait(tr.getReadVersion());
+				(void)version;
+				++self->publicNonTenantRequestPositive;
+				return Void();
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+		}
+	}
+
+	ACTOR Future<Void> testTLogReadDisallowed(Database cx, AuthzSecurityWorkload* self) {
+		state Transaction tr(cx, self->tenant);
+		state Key key = self->randomString();
+		state Value value = self->randomString();
+		self->setAuthToken(tr, self->signedToken);
+		wait(self->setKeyValue(&tr, key, value));
+		state Version committedVersion = tr.getCommittedVersion();
+		self->setAuthToken(tr, self->signedToken);
+		state Optional<Value> tLogConfigString;
+		loop {
+			try {
+				Optional<Value> value = wait(tr.get(self->tLogConfigKey));
+				ASSERT(value.present());
+				tLogConfigString = value;
+				break;
+			} catch (Error& e) {
+				wait(tr.onError(e));
+			}
+		}
+		ASSERT(tLogConfigString.present());
+		state LogSystemConfig logSystemConfig =
+		    ObjectReader::fromStringRef<LogSystemConfig>(tLogConfigString.get(), IncludeVersion());
+		state std::vector<TLogInterface> logs = logSystemConfig.allPresentLogs();
+		state std::vector<Future<TLogPeekReply>> replies;
+		for (const auto& log : logs) {
+			replies.push_back(
+			    log.peekMessages.getReply(TLogPeekRequest(committedVersion, Tag(0, committedVersion), false, false)));
```

Comment:
If the tlogs are down `getReply` could block forever. I think it would be better to use `tryGetReply` and allow `request_maybe_delivered` to be an acceptable error.

Reply:
Great suggestion. Thanks!